src/Functions/searchAnyAll.cpp

```diff
@@ -214,10 +212,18 @@ ColumnPtr FunctionSearchImpl<SearchTraits>::executeImpl(
             name,
             col_needles->getFamilyName());
 
-    if (const auto * column_string = checkAndGetColumn<ColumnString>(col_input.get()))
+    col_result->getData().resize(input_rows_count);
+    if (needles->empty())
+    {
+        /// No needles mean we don't filter and all rows pass
+        for (size_t i = 0; i < input_rows_count; ++i)
+            col_result->getData()[i] = true;
+    }
+    else
+    {
+        const auto * column_string = checkAndGetColumn<ColumnString>(col_input.get());
```

Comment:
Now that path `else if (const auto * column_fixed_string = checkAndGetColumn<ColumnFixedString>(col_input.get()))` was removed, does `searchAny/All` still support `FixedString` haystacks?

Reply:
The new tests are for FixedString needles, e.g. `WHERE searchAny(message, [toFixedString('abc', 3)])` but `col_input` refers to the haystack column. Will FixedString inputs/haystacks still work?